src/WidgetApi/Utils.js

```diff
@@ -0,0 +1,77 @@
+
+
+/**
+ * get ids of selected elements.
+ */
+export default class Utils {
+  searchIndex = null
+
+  /**
+   * constructor
+   *
+   * @param {object} searchIndex SearchIndex
+   */
+  constructor(searchIndex) {
+    this.searchIndex = searchIndex
+  }
+
+  /**
+   * get ids of selected elements.
+   *
+   * @param {object} state
+   * @return {string[]} array of GlobalIds.
+   */
+  getSelectedElementIds(state) {
+    const elementIds = []
+    // console.log(state.selectedElements)
+    if (state.selectedElements === null || state.selectedElements.length === 0) {
+      return elementIds
+    }
+
+    for (const expressId of state.selectedElements) {
+      const globalId = this.searchIndex.getGlobalIdByExpressId(expressId)
+      if (globalId) {
+        elementIds.push(globalId)
+      }
+    }
+    return elementIds
+  }
+
+  /**
+   * check if state has changed.
+   *
+   * @param {object} state
+   * @param {string[]} lastSelectedElementIds
+   * @return {boolean}
+   */
+  selectedElementIdsHasChanged(state, lastSelectedElementIds) {
+    if (Array.isArray(state.selectedElements)) {
+      const newElements = this.getSelectedElementIds(state)
+      return !this.arraysAreEqual(lastSelectedElementIds, newElements)
+    }
+  }
+
+
+  // @source https://github.com/30-seconds/30-seconds-blog/blob/master/blog_posts/javascript-array-comparison.md
+  /**
+   * check if two arrays are equal.
+   *
+   * @param {object[]} a
+   * @param {object[]} b
+   * @return {boolean}
+   */
+  arraysAreEqual(a, b) {
```

Comment:
Hm, this name is confusing given the use case.

This is a common enough utility for the whole project.  Would you please move it into src/utils/sets.js

Also I think it's worth separating out into a primary and helper
```
/**
 * @param {Array} a
 * @param {Array} b
 * @return {boolean} if sets are the same
 */
export function setsFromArraysAreEqual(a, b) {
  return setsAreEqual(new Set(a), new Set(b))
}

/**
 * @param {Set} a
 * @param {Set} b
 * @return {boolean} if sets are the same
 */
export function setsAreEqual(a, b) {
  // https://stackoverflow.com/questions/31128855/comparing-ecma6-sets-for-equality
  return a.size === b.size && [...a].every((x) => b.has(x))
}
```

Here's a preferable arraysAreEqual if that's what you want:
```
arraysAreEqual(a, b) {
  if (a.length !== b.length) {
    return false
  }
  for (let i = 0, n = a.length; i < n; i++) {
    if (a[i] !== b[i]) {
      return false
    }
  }
  return true
}
```

Reply:
Yeah you are right, the element ids lists aren't expected to have duplicates at the end. But you provided implementation requires the two arrays to be of the same order to consider equal, which is not what we seek here.

Reply:
@pablo-mayrgundter, I introduced a minimized/refactored implementation that should cover the current use case of comparing lists of element ids.

Reply:
"Momentum" model are removed from e2e tests as well.